witness/src/Dscp/DB/Rocks/Real/Functions.hs

```diff
@@ -46,28 +77,64 @@ closeNodeDB = closeRocksDB . _rdDatabase
 -- Reading/writing
 ------------------------------------------------------------
 
--- | Read ByteString from RocksDb using given key.
-rocksGetBytes :: MonadIO m => ByteString -> DB -> m (Maybe ByteString)
-rocksGetBytes key DB {..} = Rocks.get rocksDB rocksReadOpts key
+get :: (HasRocksDB ctx m, Serialise a, Serialise b) => a -> m (Maybe b)
+get a = do
+    db  <- getDB
+    mbs <- rocksGetBytes db (serialise' a)
+    case mbs of
+        Just it -> case deserialiseOrFail (LBS.fromStrict it) of
```

Comment:
Oh my god, wait, this function is just 
```
either (throwM . DeserialisationError) pure $ 
    deserialiseOrFail . LBS.fromStrict <$> rocksGetBytes db (serialise' a)
```
Pls do not make things more complex than they are. Also, redundant do at the line 89.

Reply:
@volhovm your example doesn't typecheck; this works:
```
traverse (leftToThrow DeserialisationError . deserialiseOrFail') $ rocksGetBytes db (serialise' a)
````
Frankly, I think that it's either `case` or `traverse` here, because you need to swap `Maybe` and `Monad m` somehow.